llvm/include/llvm/ProfileData/MemProf.h

```diff
@@ -306,6 +318,9 @@ struct IndexedAllocationInfo {
       if (Other.CallStack[J] != CallStack[J])
         return false;
     }
+
+    if (Other.CSId != CSId)
```

Comment:
If CSId should (already) be in sync with the CallStack, can the above CallStack comparisons be removed now?

Reply:
Yes, we can remove the `CallStack` comparison.  I was planning to remove it when we remove the `CallStack` field, but we can do so now.